libcloud/compute/drivers/kubevirt.py

```diff
@@ -0,0 +1,996 @@
+                    'https://kubernetes.io/docs/concepts/cluster-administration/certificates/'
+                )
+
+            self.cert_file = cert_file
+
+    def add_default_headers(self, headers):
+        if 'Content-Type' not in headers:
+            headers['Content-Type'] = 'application/json'
+        return headers
+
+
+class KubernetesTokenAuthentication(ConnectionKey):
+    responseCls = KubernetesResponse
+    timeout = 60
+
+    def add_default_headers(self, headers):
+        if 'Content-Type' not in headers:
+            headers['Content-Type'] = 'application/json'
+        if self.key:
+            headers['Authorization'] = 'Bearer ' + self.key
+        else:
+            raise ValueError("Please provide a valid token in the key param")
+        return headers
+
+
+class KubeVirtNode(Node):
+
+    def start_node(self):
+        self.driver.ex_start_node(self)
+
+    def stop_node(self):
+        self.driver.ex_stop_node(self)
+
+
+class KubeVirtNodeDriver(NodeDriver):
+    type = Provider.KUBEVIRT
+    name = "kubevirt"
+    website = 'https://www.kubevirt.io'
+    connectionCls = KubernetesConnection
+
+    NODE_STATE_MAP = {
+        'pending': NodeState.PENDING,
+        'running': NodeState.RUNNING,
+        'stopped': NodeState.STOPPED
+    }
+
+    def __init__(self, key=None, secret=None, secure=True, host="localhost",
+                 port=6443, key_file=None, cert_file=None, ca_cert='',
+                 token_bearer_auth=False, verify=True):
+
+        libcloud.security.VERIFY_SSL_CERT = verify
+        if token_bearer_auth:
+            self.connectionCls = KubernetesTokenAuthentication
+            if not key:
+                raise ValueError("The token must be a string")
+            secure = True
+
+        if key_file:
+            self.connectionCls = KubernetesTLSConnection
+            self.key_file = key_file
+            self.cert_file = cert_file
+            secure = True
+
+        if host.startswith('https://'):
+            secure = True
+
+        # strip the prefix
+        prefixes = ['http://', 'https://']
+        for prefix in prefixes:
+            if host.startswith(prefix):
+                host = host.lstrip(prefix)
+
+        super(KubeVirtNodeDriver, self).__init__(key=key,
+                                                 secret=secret,
+                                                 secure=secure,
+                                                 host=host,
+                                                 port=port,
+                                                 key_file=key_file,
+                                                 cert_file=cert_file)
+
+        # check if both key and cert files are present
+        if key_file or cert_file:
+            if not(key_file and cert_file):
+                raise Exception("Both key and certificate files are needed")
+
+        if ca_cert:
+            self.connection.connection.ca_cert = ca_cert
+        else:
```

Comment:
Since ``ca_cert`` is set to None by default it might not be a bad idea to emit a warning if ca cert is not used that server SSL certificate won't be validated against the CA cert.

And when the documentation is added, it would also be good to point that out there.